pygsheets/client.py

```diff
@@ -74,152 +75,81 @@ def __init__(self, oauth, http_client=None, retries=1, no_cache=False):
         data_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
         with open(os.path.join(data_path, "sheets_discovery.json")) as jd:
             self.service = discovery.build_from_document(jload(jd), http=http)
-        with open(os.path.join(data_path, "drive_discovery.json")) as jd:
-            self.driveService = discovery.build_from_document(jload(jd), http=http)
+        self.drive = DriveAPIWrapper(http, data_path)
         self._spreadsheeets = []
         self.batch_requests = dict()
         self.retries = retries
         self.enableTeamDriveSupport = False  # if teamdrive files should be included
         self.teamDriveId = None  # teamdrive to search for spreadsheet
-        self._spreadsheeets = self._fetch_sheets()
 
-    def _fetch_sheets(self, filter_query=""):
-        """
-        fetch all the sheets info from user's gdrive
+    def spreadsheet_ids(self, query=None):
+        """A list of all the ids of spreadsheets present in the users drive or TeamDrive."""
+        return [x['id'] for x in self.drive.spreadsheet_metadata(query)]
 
-        :returns: None
-        """
-        if filter_query != "":
-            filter_query = " and " + filter_query
-        request = self.driveService.files().list(corpora='teamDrive' if self.teamDriveId else 'user', pageSize=500,
-                                                 fields="files(id, name)", teamDriveId=self.teamDriveId,
-                                                 q="mimeType='application/vnd.google-apps.spreadsheet'"+filter_query,
-                                                 supportsTeamDrives=self.enableTeamDriveSupport,
-                                                 includeTeamDriveItems=self.enableTeamDriveSupport)
-        results = self._execute_request(None, request, False)
-        try:
-            results = results['files']
-        except KeyError:
-            results = []
-        return results
+    def spreadsheet_titles(self, query=None):
+        """A list of all the titles of spreadsheets present in the users drive or TeamDrive."""
+        return [x['name'] for x in self.drive.spreadsheet_metadata(query)]
```

Comment:
why not cache the spreadsheet metadatas ? like it was done using  `_spreadsheets`? As there is a less chance for the spreadsheet's metadata's (at-least title and id to be changed) while script is running, and if its changed then update the cache (it is implemented in previous code).

Reply:
I changed this because the Drive API does not count towards the same API limitations as Sheet API calls. So there is no need to limit the number of calls. And making the request does not take long at all. And is only very rarely called in normal operation. 

This way ensures that the data is always up to date. In the old implementation it was up to the user to realize when to update the metadata. (it would try to load new data when a sheet was not found or when no sheets were loaded yet.) 

So overall I do not really see the advantage of the old implementation.

Reply:
Ahh, its Cool then.